label_sync/README.md

```diff
@@ -1,55 +1,71 @@
-You can run it locally with repositories and labels data fetched into YAML files:
+# label_sync
 
-Run (with go):
-`go run label_sync/main.go -help` (for help)
-`go run label_sync/main.go -labels-path label_sync/labels.yaml -local -repos label_sync/repos.yaml -repo-labels label_sync/repo_labels.yaml`
+Update or migrate github labels on repos in a github org based on a YAML file
 
-Test (with go):
-`go test k8s.io/test-infra/label_sync`
-`make test`
+## Configuration
 
-Bazel test:
-`bazel test //label_sync/...`
+A typical labels.yaml file looks like:
 
-Bazel build:
-`bazel build //label_sync/...`
+```yaml
+---
+labels:
+  - color: 000ff
+    name: lgtm
+  - color: ff0000
+    name: priority/P0
+    previously:
+    - color: 00ff00
+      name: P0
+  - name: dead-label
+    color: cccccc
+    deleteAfter: 2017-01-01T13:00:00Z
+```
 
-Bazel run:
-`./bazel-bin/label_sync/label_sync -labels-path bazel-test-infra/label_sync/labels.yaml -local -repos bazel-test-infra/label_sync/repos.yaml -repo-labels bazel-test-infra/label_sync/repo_labels.yaml`
+This will ensure that:
 
-Build static binary:
-`make compile-static`
+- there is a green `lgtm` label
+- there is a red `priority/P0` label, and previous labels should be migrated to it:
+  - if a `P0` label exists:
+    - if `priority/P0` does not, modify the existing `P0` label
+    - if `priority/P0` exists, `P0` labels will be deleted, `priority/P0` labels will be added
+- the `dead-label` label will be deleted after 2017-01-01T13:00:00Z
```

Comment:
Note: we will not delete labels until after 2017, but we also won't create the label if it doesn't exist https://github.com/spiffxp/test-infra/blob/66528d0eebb66e15e8ee76d82c6c62227b355032/label_sync/main.go#L241